packages/office-ui-fabric-react/src/components/Button/BaseButton.tsx

```diff
@@ -580,12 +586,44 @@ export class BaseButton extends BaseComponent<IBaseButtonProps, IBaseButtonState
     );
   }
 
-  private _onMouseDown = (ev: React.MouseEvent<BaseButton>) => {
-    if (this.props.onMouseDown) {
-      this.props.onMouseDown(ev);
+  private _onKeyDown = (ev: React.KeyboardEvent<HTMLDivElement | HTMLAnchorElement | HTMLButtonElement>) => {
+    // explicity cancelling event so click won't fire after this
+    if (this.props.disabled && (ev.which === KeyCodes.enter || ev.which === KeyCodes.space)) {
+      ev.preventDefault();
+      ev.stopPropagation();
+    } else if (!this.props.disabled && this.props.onKeyDown !== undefined) {
+      this.props.onKeyDown(ev); // not cancelling event because it's not disabled
+    }
+  };
+
+  private _onKeyUp = (ev: React.KeyboardEvent<HTMLDivElement | HTMLAnchorElement | HTMLButtonElement>) => {
+    if (!this.props.disabled && this.props.onKeyUp !== undefined) {
+      this.props.onKeyUp(ev); // not cancelling event because it's not disabled
+    }
+  };
+
+  private _onKeyPress = (ev: React.KeyboardEvent<HTMLDivElement | HTMLAnchorElement | HTMLButtonElement>) => {
+    if (!this.props.disabled && this.props.onKeyPress !== undefined) {
+      this.props.onKeyPress(ev); // not cancelling event because it's not disabled
     }
+  };
+
+  private _onMouseUp = (ev: React.MouseEvent<HTMLDivElement | HTMLAnchorElement | HTMLButtonElement>) => {
+    if (!this.props.disabled && this.props.onMouseUp !== undefined) {
+      this.props.onMouseUp(ev); // not cancelling event because it's not disabled
+    }
+  };
 
-    ev.preventDefault();
+  private _onMouseDown = (ev: React.MouseEvent<HTMLDivElement | HTMLAnchorElement | HTMLButtonElement>) => {
+    if (!this.props.disabled && this.props.onMouseDown !== undefined) {
+      this.props.onMouseDown(ev); // not cancelling event because it's not disabled
+    }
+  };
+
+  private _onClick = (ev: React.MouseEvent<HTMLDivElement | HTMLAnchorElement | HTMLButtonElement>) => {
+    if (!this.props.disabled && this.props.onClick !== undefined) {
+      this.props.onClick(ev); // not cancelling event because it's not disabled
+    }
```

Comment:
I'm a bit concerned about all of the explicit filtering.. did this work before? Why is this now needed? I'm wondering if a previously set 'disabled' properly is now not being updated properly as root cause.

Reply:
This is needed exactly because allowDisabledFocus makes it to where the native HTML disabled attribute is not set (otherwise buttons would not be focusable). When rendering like this, aria-disabled is set to still convey that the control is disabled. One side effect of not using HTML disabled is that all events can still fire against the button (where they don't with the disabled HTML attribute).

Reply:
With these wrappers, the callbacks should get called the same regardless of the allowDisabledFocus value

Reply:
Got it, thanks